libsolidity/grammar.txt

```diff
@@ -79,10 +79,12 @@ Expression =
 
 PrimaryExpression = Identifier | BooleanLiteral | NumberLiteral | StringLiteral
 
-FunctionCall = Identifier '(' Expression? ( ',' Expression )* ')'
+FunctionCall = ( MemberAccessFunctionCall | IndexAccessFunctionCall ) '(' Expression? ( ',' Expression )* ')'
```

Comment:
They can also be mixed (`a.b[5].c()`), so I think it should  be something like

`FunctionCall = ( PrimaryExpression | NewExpression | TypeName ) (  ( '.'  Identifier ) | ( '[' Expression ']' ) )*